### DISCUSSION TITLE
Unique jobs

### DISCUSSION THREAD

**Discussion by till** [riverqueue/river#553] [Q&A] [answered]
Hey 👋🏼 !

First off, congrats on building this, looks super nice — docs are great. :) And the Pro option looks neat as well.

I had a question about unique jobs, and how to maintain them.

My idea was that I'd have a client would submit my "unique" jobs every time the process starts. The jobs are of a cronjob-like nature (let's say a key-rotation thing that I want to run every 3 months) and I want only one to be running. But instead of writing a one-of-script to create them, I would do it in my app startup to have a place where I maintain them.

Is this currently possible? Or would I be better off with a one of script that takes care of them?

**Comment by brandur** [comment]
@till Thanks! Did you see the `RunOnStart` option?

https://riverqueue.com/docs/periodic-jobs#basic-usage

I'm not sure I totally grok what you want here, but are you asking whether River would be able to manage and guarantee the once every three months schedule? If so, no it definitely can't do that at the moment.

**Reply by till** [reply]
Hey — thanks for your reply, and sorry, I didn't explain it correctly.

So I am aware of periodic jobs and how they work, what I am wondering is, if I can make them unique.

So the idea is two fold:

- periodic
- only one at the time

I want to add the jobs to a start-up sequence of one of my services, so when it boots, it will try to register them or update them to river, but if they exist already, it will just continue (without error).

Goal was to "maintain" them in "code" vs. writing one-off go "scripts" to register the jobs with river.

**Reply by brandur** [reply, marked as answer]
Ah gotcha. You can combine periodic jobs with unique jobs, and unique jobs have a `ByPeriod` argument that lets you specify the period over which they're unique. That said though, it doesn't have a perfectly granular schedule for such. It's a duration like "unique every 15 minutes" or "unique every hour".

**Reply by till** [reply]
Alright, yeah, it doesn't need to be perfect, just needs to happen. :D I'll tinker with it. Thank you!
